sway/tree/view.c

```diff
@@ -844,6 +844,7 @@ void view_unmap(struct sway_view *view) {
 		seat_consider_warp_to_focus(seat);
 	}
 
+	wl_signal_emit(&view->container->node.events.destroy, &view->container->node);
```

Comment:
I'm not very comfortable moving the `wl_signal_emit` call here. This makes it so the signal is no longer emitted for containers which aren't views.

Is it possible to emit it in `container_destroy` instead?

Reply:
Unfortunately emitting the signal any later segfaults in cursor_rebase. I think it doesn't expect view->surface == NULL.